Dockerfile

```diff
@@ -224,6 +224,38 @@ RUN apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(ls
   && rm -rf /var/lib/apt/lists/*
 
 
+### DART
+
+# Install Dart 
+ENV PUB_CACHE=/opt/dart/pub-cache \
+  PUB_ENVIRONMENT="dependabot" \
+  PATH="${PATH}:/opt/dart/dart-sdk/bin"
+ARG DART_VERSION=2.14.2
+RUN curl --connect-timeout 15 --retry 5 "https://storage.googleapis.com/dart-archive/channels/stable/release/${DART_VERSION}/sdk/dartsdk-linux-x64-release.zip" > "/tmp/dart-sdk.zip" \
+  && mkdir -p "$PUB_CACHE" \
+  && chown dependabot:dependabot "$PUB_CACHE" \
+  && unzip "/tmp/dart-sdk.zip" -d "/opt/dart" > /dev/null \
+  && chmod -R o+rx "/opt/dart/dart-sdk" \
+  && rm "/tmp/dart-sdk.zip" \
+  && dart --version
+# HACK: We use a patched version of "dart pub" that is not in the Dart SDK yet.
+#       Hence, we clone from git and globally install it from local folder.
+#       Now it can be used with: 'dart pub global run pub ...'
+RUN git clone https://github.com/dart-lang/pub.git /opt/dart/pub \
+  && git -C /opt/dart/pub checkout dependency_services \
+  && dart pub global activate --source path /opt/dart/pub \
+  && chmod -R o+r "/opt/dart/pub" \
+  && chown -R dependabot:dependabot "$PUB_CACHE" \
+  && chown -R dependabot:dependabot /opt/dart/pub
+
+# Install Flutter
+ARG FLUTTER_VERSION=2.5.3
+RUN curl --connect-timeout 15 --retry 5 "https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_${FLUTTER_VERSION}-stable.tar.xz" > "/tmp/flutter.xz" \
+  && tar xf "/tmp/flutter.xz" -C /opt/dart \
+  && chmod -R o+rx "/opt/dart/flutter" \
+  && chown -R dependabot:dependabot "/opt/dart/flutter" \
+  && runuser -l dependabot -c "/opt/dart/flutter/bin/flutter --version"
```

Comment:
Do y'all have any indication what this means for the size of the resulting docker image? It's already fairly large, and we might want to start thinking about splitting things out (which will require some work on our end)

Reply:
Locally on my linux machine this expands to a 831M folder.

If we wanted to try to optimize for space:

The only things we need from flutter is:
* the version
* the "sdk-packages" (< 50 megs)

If we could obtain those  'out-of-band' we could use the dart-sdk only (~481M decompressed)... Not sure it is worth it...

Reply:
Or we could delete everything else after extracting the Flutter SDK :D

Reply:
I built this and found that these additions increased the image size by about 3GB which is almost doubling the size. If it's possible to optimize this it would be very helpful as we're already concerned about the size of the dependabot-core image and would need to look into splitting the image by ecosystem to accommodate that much growth.

The dart sdk was around 600MB:
```
/bin/bash -o pipefail -c curl --connect-timeout 15 --retry 5 "https://storage.googleapis.com/dart-archive/channels/stable/release/${DART_VERSION}/sdk/dartsdk-linux-x64-release.zip" > "/tmp/dart-sdk.zip"   && mkdir -p "$PUB_CACHE"   && chown dependabot:dependabot "$PUB_CACHE"   && unzip "/tmp/dart-sdk.zip" -d "/opt/dart" > /dev/null   && chmod -R o+rx "/opt/dart/dart-sdk"   && rm "/tmp/dart-sdk.zip"   && dart --version # buildkit                                                                                                                    573MB
```

Flutter was around 2.3GB
```
/bin/bash -o pipefail -c curl --connect-timeout 15 --retry 5 "https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_${FLUTTER_VERSION}-stable.tar.xz" > "/tmp/flutter.xz"   && tar xf "/tmp/flutter.xz" -C /opt/dart   && chmod -R o+rx "/opt/dart/flutter"   && chown -R dependabot:dependabot "/opt/dart/flutter"   && runuser -l dependabot -c "/opt/dart/flutter/bin/flutter --version" # buildkit                                                                                                    2.33GB
```

Ideally we wouldn't want to add more than a few hundred MB to our current monolithic image if we were to include pub support. Would it be possible to trim it down that much?